python/sedUtils.py

```diff
@@ -0,0 +1,273 @@
+    If redshift is a dictionary containing numpy arrays in the 'quasar', 'sn' or 'galaxies' keys, then
+    those redshifts are applied to those objects only. """
+    # Check what kind of redshift object we received.
+    if isinstance(redshifts, dict):
+        # Loop over all the object types in the redshift dictionary.
+        for objtype in redshifts.keys():
+            # Then loop over all redshifts for this type.
+            newSedsForList = []
+            for z in redshifts[objtype]:
+                # And add new SEDs redshifted to this z for this type of object.
+                for s in sedlists[objtype]:
+                    # Make the new name for the new SED at this redshift.
+                    newsedname = s + '_Z_%.3f' %(z)
+                    # Add it to the SED dictionary.
+                    sedDict[newsedname] = redshiftSingleSED(sedDict[s], z)
+                    newSedsForList.append(newsedname)
+            # Done redshifting all objects of this type; updated sedlist.
+            sedlists[objtype] += newSedsForList
+    else:
+        # Using the same redshift list for everything (except stars).
+        for z in redshifts:
+            for objtype in sedlists.keys():
+                newSedsForList = []
+                if objtype == 'stars':
+                    # Skip stars.
+                    continue
+                for s in sedlists[objtype]:
+                    newsedname = s + '_Z_%.3f' %(z)
+                    sedDicts[newsedname] = redshiftSingleSED(sedDict[s], z)
+                    newSedsForList.append(newsedname)
+                sedlists[objtype] += newSedsForList
+    return sedDict, sedlists
+
+
+def redshiftSingleSED(sed_in, z):
+    # Make a copy of the input SED, as otherwise we will overwrite original arrays.
+    sed_out = deepcopy(sed_in)
+    sed_out.redshiftSED(z, dimming=False)
+    return sed_out
+
+
+def matchSedsBp(sedDict, bpDict, refFilter=None):
+    """Match the wavelength ranges for all the Seds and the bandpass dictionary.
+    This will speed up later calculation of magnitudes (if you're doing a lot of them),
+     but is not strictly necessary. """
+    if refFilter == None:
+        refFilter = bpDict.keys()[0]
+    wavelen_match = bpDict[refFilter].wavelen
+    # Check all filters in bpDict match in wavelength space (note bandpasses must be regular grid).
+    for f in bpDict.keys():
+        if numpy.any(bpDict[f].wavelen != wavelen_match):
+            bpDict[f].resampleBandpass(wavelen_min=wavelen_match.min(), wavelen_max=wavelen_match.max(),
+                                       wavelen_step = wavelen_match[1] - wavelen_match[0])
+    # Check all seds in sedDict match in wavelength space.
+    for s in sedDict.keys():
+        if sedDict[s].needResample(wavelen_match=wavelen_match):
+            sedDict[s].resampleSED(wavelen_match=wavelen_match)
+    return sedDict, bpDict
+
+
+def calcNatMags(bandpassDict, sedDict, sedlists):
```

Comment:
wait.  Nevermind.  I see it.  I am very confused what the difference between sedDict and sedLists is.